### ISSUE TITLE
Need a .generatedNoFiles() proposition.

### ISSUE THREAD

**Issue by cgruber** [google/compile-testing#15]
This is important both in terms of asserting that a compiltion run doesn't produce files when it isn't supposed to, but could also in the error list all generated files in that compile job, and be useful to debug the processor, since that list is hard to know/discover without going to the filesystem and digging.

**Comment by blipinsk** [comment]
:+1: especially with the option to assert if one specific file hasn't been generated (not only that nothing has been generated)

**Comment by tokazio** [comment]
+1 for this functionnality. I want to test that in a specified test case, my annotation processor does not generate any file.

**Comment by eamonnmcmanus** [comment]
I think this bug is obsolete. The current API includes `Compilation.generatedFiles()` and you can easily assert the returned list is empty.

**Comment by tokazio** [comment]
ok sorry. Then the version should be 0.11 in the read me ;)

**Comment by tbroyer** [comment]
READMEs (in general) should probably always link to the GitHub [releases page](https://github.com/google/compile-testing/releases) for the latest release (with release notes), [search.maven.org](https://search.maven.org/#search%7Cgav%7C1%7Cg%3A%22com.google.testing.compile%22%20AND%20a%3A%22compile-testing%22) for how to declare a dependency (possibly using https://github.com/jirutka/maven-badges to directly link to the latest release directly), and [javadoc.io](https://javadoc.io/doc/com.google.testing.compile/compile-testing) for javadoc.

**Comment by cgruber** [comment]
Yeah - we need to clean a few of those things up.  I'll try to get to it
this week.

**Comment by JakeWharton** [comment]
Still needed in the clauses

**Comment by ronshapiro** [comment]
`assertThat(compilation.generatedSourceFiles()).isEmpty()` should be sufficient now. We can add a method onto `CompilationSubject` if we think that would be helpful, but I'm not sure it would be used enough to merit it.

**Comment by ZacSweers** [comment]
I think it'd be fairly useful for either processors that only validate/conditionally generate code, or for error cases where you'd want to assert that errors are reported eagerly and no code gen is run. For instance - I'd make this basically a standard part of all error checks in processors I've written
